client-umb/client.async.py

```diff
@@ -31,64 +33,71 @@
 
     # Opening WebSocket Connection to WebSocket server
     async with websockets.connect(ws_endpoint) as websocket:
-        logging.warning('--- Opened Connection to Server ---')
+        logging.info('--- Opened Connection to Server ---')
         # sending websocket message to server
         await websocket.send(json.dumps({'msg':'TEST MESSAGE'}))
         while True:
             # Listening for recived messages from websocket server
             response = await websocket.recv()
+            # Connection from websocket server is closed so terminate this iterations
+            if response is None:
+                logging.warning("Response None. Exit.")
+                break
             logRecivedMessage(response) # Log websocket recived message
             data = json.loads(response)
-            payload = {}
-            payload["payload"] = json.loads(data["payload"])
-            logRecivedMessagePayload(payload["payload"]) # log websocket message payload from webhook
-            # REF Github branch for building project
-            ref = payload["payload"]["ref"].split("/")[2]
+            message = {}
+            payload = json.loads(data["payload"])
+            message["payload"] = payload
+            logRecivedMessagePayload(payload)
             # Create Dictionary of allowed headers from rerouted webhook POST request
-            payload['headers'] =  {x:y for x,y in data["headers"].items() if x in alowed}
-            # Connection from websocket server is closed so terminate this iterations
-            if response is None: break
+            message['headers'] =  {x:y for x,y in data["headers"].items() if x in alowed}
+            if "ref" in payload:
+                # REF Github branch for building project
+                ref = payload["ref"].split("/")[2]
             else:
-                if branch_name != '' and ref in branch_name.split(","):
-                    amqp_props = dict()
-                    amqp_props['subject'] = payload['payload']['head_commit']['message']
-                    amqp_message = str(payload)  # "Test Message"
-                    producer = AMQProducer(
-                        urls=amqp_url,
-                        certificate=rh_crt,
-                        private_key=rh_key,
-                        trusted_certificates=rh_cert,
-                        topic=topic+amqp_topic
-                    )
-                    # send AMQP message to Red Hat UMB service
-                    producer.send_msg(amqp_props,amqp_message.encode("utf-8"))
-                else:
-                    logDifferentBranchName(ref,branch_name)
-
+                logging.info("Not containing ref")
+                continue
+            # If not specify branches, we accept all
+            if branch_name == '' or ref in branch_name.split(","):
```

Comment:
Problem with not adding branch name  ( or names ) is that our building process or pipeline will be triggered on every single change from every project that will be working on and that from my opinion is very bad because different project builds will be triggered for no reason at every time someone else is making some other change on some other project.

Reply:
It looks to me the branch_name is just "master" or something like "master,1.8.x". TBO, I don't think it makes much sense. As the client serves many repositories, we can not set it properly or simply to "master" or "1.8.x" since we have other projects, and we have other branches, e.g. "1.9/10/11...x". Maybe I miss something?

Reply:
I got yor point with "master" and that was my concern also but then i think that we can just append project name before branch name ( project name we would enter on env. variable with some separator like "." or "/"  and then we we get payload from hooks we can check that value with provided list of project["."]branch names and make our logic for different projects in that way. 
Leaving "ref" to be empty will be trigger for every code change that we get from github hooks . If we have only one project and one developer that would be working on , it is fine , he will know when he has made change and that in same time rebuild of new service with that code will be redeployed, but imagine that two developers are working on different repositories in same time...  then services will be redeployed on every change from every developer and this is more bad if they work on same codebase but expect to see their code changes redeployed in new service... i don't wanna think what would start to happen when three or more starting working on two or more same or different projects :) .
In any way , maybe best solution would be to test it in real case scenario and we all to see how it will perform.

Reply:
Yeah, we can use a pattern or something to filter the messages. 
Actually, my code only changes one semantics. It is now "if you do not specify a branch, I will allow any branch", rather than "if you do not specify a branch, I will deny all" (old code).